webapp-django/crashstats/tokens/middleware.py

```diff
@@ -65,6 +65,7 @@ def process_request(self, request):
         # we use as long as it's something
         user.backend = "django.contrib.auth.backends.ModelBackend"
         user.has_perm = partial(has_perm, token.permissions.all())
+        user.token_login = True
```

Comment:
This allows us to differentiate between whether this "user" logged in via normal login or via an auth token.